### PR TITLE
17047 migrate PressableWithSecondaryInteraction to PressableWithFeedback component

### PR DESCRIPTION
**Pull request by robertKozik**
### Fixed Issues

$ https://github.com/Expensify/App/issues/17047
PROPOSAL: https://github.com/Expensify/App/issues/17047

### Tests

PressableWithSecondaryInteraction is used in three places: reports in LHN, options in RHN, and emoji bubbles and links/mails in Chats
1. LHN
    - log in
    - verify that styling of each individual report visible in LHN is unchanged
    - longPress(touchable devices)/ rightClick (desktop) the report and verify that popover is opened correctly
    - press/click the report and verify that user is directed to this report correctly
2. RHN
    - log in
    - click the profile avatar in LHN
    - verify that each option visible on opened RHN has unchanged styling
    - press/click and verify that menu item redirects correctly
3. ReportScreen
    - log in
    - navigate to chat with another user
    - type and send any link
    - verify that styling of the link remain unchanged
    - longPress(touchable devices)/ rightClick (desktop) the link and verify that popover is opened correctly
    - add reaction to message
    - longPress(touchable devices)/ rightClick (desktop) reacted emoji and verify that popover is opened correctly

- [X] Verify that no errors appear in the JS console

### QA Steps

- [X] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/61146594/53011e0f-5c0f-4f66-a4e1-b922e12b8e62

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/61146594/5dd25178-a349-4e5c-9f72-02f1a4f79eeb

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/61146594/2639e06e-0960-4cae-be2f-89482d784b6d

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/61146594/510a66a5-493c-419f-9790-1b12bcc85b80

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/61146594/baa5eb23-1500-4be1-b536-e8e43780fbee

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/61146594/d7af3776-7931-4a2f-b25f-fa309430f399

</details>

### CODE REVIEW

File: src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js [hunk 1]
```diff
@@ -60,12 +60,13 @@ const BaseAnchorForCommentsOnly = (props) => {
             onPress={linkProps.onPress}
             onPressIn={props.onPressIn}
             onPressOut={props.onPressOut}
+            accessibilityRole="link"
```
Comment (luacmartins) [inline src/components/AnchorForCommentsOnly/BaseAnchorForCommentsOnly.js:63]: NAB should we create a CONST for `link`?

**Review by 0xmiroslav — COMMENTED** [review]
@robertKozik please link correct issue. It's not 14047 but #17047

File: src/components/LHNOptionsList/OptionRowLHN.js [hunk 2]
```diff
@@ -144,10 +144,12 @@ const OptionRowLHN = (props) => {
                             props.isFocused ? styles.sidebarLinkActive : null,
                             hovered && !props.isFocused ? props.hoverStyle : null,
                         ]}
+                        accessibilityRole="button"
```
Comment (luacmartins) [inline src/components/LHNOptionsList/OptionRowLHN.js:147]: Same here

File: src/components/MenuItem.js [hunk 3]
```diff
@@ -122,6 +122,8 @@ const MenuItem = (props) => {
             ]}
             disabled={props.disabled}
             ref={props.forwardedRef}
+            accessibilityRole="menuitem"
```
Comment (luacmartins) [inline src/components/MenuItem.js:125]: Same here

**Review by luacmartins — DISMISSED** [review]

**Comment by robertKozik** [comment]
Created the CONST values for accessibility roles 🚀

**Review by luacmartins — DISMISSED** [review]
LGTM! Thanks for adding the const.

**Comment by luacmartins** [comment]
@0xmiroslav all yours

**Review by luacmartins — DISMISSED** [review]

**Comment by luacmartins** [comment]
@robertKozik did you manage to reproduce the issues @0xmiroslav mentioned above?

**Comment by robertKozik** [comment]
I'm on it right now

**Comment by robertKozik** [comment]
@0xmiroslav  I cannot reproduce the problem. Can you provide me more context? I'm assuming that long pressing over the text should cause the issue, but I can only encounter the problem from production.

What can be done, to assure that no text can be selected is to add `styles.userSelectNone` and `styles.noSelect` for the styling of PressableWithSecondaryInteraction in OptionRowLHN. But it isn't solving the problem from the prod.

**Comment by robertKozik** [comment]
@0xmiroslav @luacmartins I had difficulties with reproduction the problem on mSafari, but it turned out that it is more visible on the mChrome. When user tried to longPress over the text, selection of it has priority over our defined handler. I came up with the following solution:
_When we are on touch device, selection of the text inside `PressableWithSecondaryInteraction` is disabled by `styles.userSelectNone` and `styles.noSelect`_. 

The problem from the prod is still visible tho, it only occurs on the mSafari. With above styles applied, it's less intrusive, but still visible.

**Comment by robertKozik** [comment]
PressableWithSecondaryInteraction is only used in three places, which are listed here. If my assumptions of the nature of the problem are correct (text selection), it can occur on every instance of it when it wraps text.

**Comment by luacmartins** [comment]
@robertKozik we have conflicts.

File: src/components/Reactions/EmojiReactionBubble.js [hunk 4]
```diff
@@ -59,6 +59,8 @@ function EmojiReactionBubble(props) {
             enableLongPressWithHover={props.isSmallScreenWidth}
             // Prevent text input blur when emoji reaction is clicked
             onMouseDown={(e) => e.preventDefault()}
+            accessibilityRole="button"
```
Comment (luacmartins) [inline src/components/Reactions/EmojiReactionBubble.js:62]: Let's use the new CONST.

**Review by luacmartins — CHANGES_REQUESTED** [review]

File: src/components/Reactions/EmojiReactionBubble.js (hunk 4 above)
Comment (robertKozik) [reply src/components/Reactions/EmojiReactionBubble.js]: done

**Comment by robertKozik** [comment]
@0xmiroslav bumping, as I said in my comment before, the no selection policy is only referring to the PressableWithSecondaryInteraction (3 places in the app). On the other hand, if you prefer, that prohibition of text selection can only be added inside the LHN instance of the PressableWithSecondaryInteraction.

**Comment by luacmartins** [comment]
@robertKozik we have conflicts. Could you please resolve? I'll do a final review once that's done.

**Comment by robertKozik** [comment]
@luacmartins merged with main

**Review by 0xmiroslav — APPROVED** [review]

**Comment by robertKozik** [comment]
> As a follow-up, I suggest to
> 
> * generalize solution to prevent selection on mWeb ([17047 migrate PressableWithSecondaryInteraction to PressableWithFeedback component #20251 (comment)](https://github.com/Expensify/App/pull/20251#discussion_r1238008612))
> * use CONST for all static values of accessibilityRole ([17047 migrate PressableWithSecondaryInteraction to PressableWithFeedback component #20251 (comment)](https://github.com/Expensify/App/pull/20251#discussion_r1224474305))

The issue for the const values is already created ([LINK](https://github.com/Expensify/App/issues/21228))

**Review by luacmartins — APPROVED** [review]

File: src/components/MenuItem.js [hunk 5]
```diff
@@ -122,6 +122,8 @@ function MenuItem(props) {
             ]}
             disabled={props.disabled}
             ref={props.forwardedRef}
+            accessibilityRole={CONST.ACCESSIBILITY_ROLE.MENUITEM}
+            accessibilityLabel={props.title}
```
Comment (abdulrahuman5196) [inline src/components/MenuItem.js:126]: props.title was of number `0` in some cases which caused app crash as mentioned here https://github.com/Expensify/App/issues/25451#issuecomment-1683356992.
We should make sure the values we pass are string.

**Pull request merged**